### PR TITLE
NAS strategy (stage 4) - Evolution, HPO and oneshot strategy

### PR DESCRIPTION
**Pull request by ultmaster**

### CODE REVIEW

File: nni/nas/strategy/hpo.py [hunk 1]
```diff
@@ -56,45 +115,8 @@ class TPE(BaseStrategy):
     """
 
     def __init__(self):
```
Comment (liuzhe-lz) [inline nni/nas/strategy/hpo.py:117]: Add `seed` parameter?

Comment (ultmaster) [reply nni/nas/strategy/hpo.py:117]: Why? How to set seed for TpeTuner?

Comment (liuzhe-lz) [reply nni/nas/strategy/hpo.py:117]: https://nni.readthedocs.io/en/stable/reference/hpo.html#nni.algorithms.hpo.tpe_tuner.TpeTuner
`TpeTuner(optimize_mode='minimize', seed=None, tpe_args=None)`

File: nni/nas/strategy/hpo.py [hunk 2]
```diff
@@ -3,98 +3,127 @@
...
-                if is_stopped_exec(_model):
-                    if _model.metric is not None:
-                        self.tpe_sampler.receive_result(_id, _model.metric)
-                        _logger.debug('tpe receive results: %d, %s', _id, _model.metric)
-                    to_be_deleted.append(_id)
-            for _id in to_be_deleted:
-                del self.running_models[_id]
+    def __init__(self, *args, **kwargs):
+        from nni.algorithms.hpo.tpe_tuner import TpeTuner
+        super().__init__(TpeTuner())
```
Comment (liuzhe-lz) [inline nni/nas/strategy/hpo.py:126]: 🙂

**Review by liuzhe-lz — APPROVED** [review]

**Review by liuzhe-lz — APPROVED** [review]

**Review by QuanluZhang — APPROVED** [review]
seems hpo tuners have been aligned in this pr, right?, which is a good news!

**Pull request merged**